src/transition-jira-tasks.ps1

```diff
@@ -1,19 +1,48 @@
 param (
     [string]$JiraDomain,
     [string]$JqlToQueryBy,
+    [string[]]$IssueKeys,
     [string]$NewState,
+    [PSCustomObject]$Fields,
+    [PSCustomObject]$Updates,
+    [string]$Comments
     [string]$JiraUsername,
     [securestring]$JiraPassword
 )
 
+$global:InformationPreference = "Continue"
 $ErrorActionPreference = "Stop"
 
 . $PSScriptRoot\transition-functions.ps1
 
 $baseJiraUri = New-Object -TypeName System.Uri -ArgumentList "https://$JiraDomain/"
 
-Invoke-JiraTransitionTickets -BaseUri $baseJiraUri `
+if ([string]::IsNullOrEmpty($JqlToQueryBy) -And $IssueKeys.length -gt 0) {
+    $JqlToQueryBy = "key IN (""$($IssueKeys -join '", ')"")"
+}
+
+$processedIssue = Invoke-JiraTransitionTickets -BaseUri $baseJiraUri `
     -Username $JiraUsername `
     -Password $JiraPassword `
     -Jql $JqlToQueryBy `
-    -Transition $NewState
\ No newline at end of file
+    -Transition $NewState `
+    -Fields $Fields `
+    -Updates $Updates `
+    -Comment $Comment
+    
+$identifiedIssueKeys = $processedIssues.Keys
+$transitionedIssueKeys = $processedIssues | Where-Object { $_.Value -eq $true } | Select-Object { $_.Key }
```

Comment:
Issues can fail when transitioning if they have certain requirements (like a specific field must be populated).